google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1310,7 +1310,7 @@ public Builder setIamConfiguration(IamConfiguration iamConfiguration) {
 
     @Override
     public Builder setLogging(Logging logging) {
-      this.logging = logging;
+      this.logging = logging != null ? logging : null;
```

Comment:
Instead do:

```
this.logging = logging != null ? logging : Data.nullOf(Bucket.Logging.class);
```

Reply:
@frankyn 
I think we have to go with the `null ` instead of `Data.nullOf(Bucket.Logging.class)` because if we do as you said then the following error occurs :
`unable to create new instance of class com.google.cloud.storage.BucketInfo$Logging because it has no accessible default constructor`